msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/ui/MessageForumStatisticsBean.java

```diff
@@ -883,6 +1030,37 @@ public List getUserAuthoredStatistics2(){
 		return statistics;
 	}
 	
+	/**
+	 * Filters out messages associated with anonymous topics for which the user is not permitted to see IDs
+	 */
+	private List<Message> filterToAnonAwareMessages(List<Message> messages)
+	{
+		if (!isAnonymousEnabled() || m_displayAnonIds)
+		{
+			return messages;
+		}
+
+		List<Message> filtered = new ArrayList<Message>();
+		// Map topics to value of 'isUseAnonymousId(Topic)' to prevent redundant queries
+		Map<Topic, Boolean> topicToUseAnon = new HashMap<Topic, Boolean>();
+		for (Message message : messages)
+		{
+			Topic topic = message.getTopic();
```

Comment:
In the context of Forums (but not Messages), message.getTopic() is never null. However, accessing the topic's attributes other than the ID may result in LazyInitializationException depending how the message was retrieved. In this case the calling methods use queries that join on the topic table, so we should be safe.

In terms of messages, the calling methods are passing in the result of Query.list(), which is never null. But some null checks etc. here would be good for extra confidence.